### PR TITLE
Fix polling for haltestellen and add iframe service

### PR DESCRIPTION
**Pull request by loicgasser**
Fix for https://github.com/geoadmin/mf-chsdi3/issues/2235 (no more polling issue and htmlPopup as a service)

This PR introduces a popup service which we call "iframeHtmlPopup"

This is service is always meant to be called within the "regular" htmlPopup service.

As a side note, if the iframeHtmlPopup is called and no template is defined for this service, it will simply return an empty html.

[Now](https://mf-chsdi3.dev.bgdi.ch/iframe_service/rest/services/ech/MapServer/ch.bav.haltestellen-oev/8573140/htmlPopup?coord=688630.0000000001,165610.00000000006&imageDisplay=1920,750,96&lang=fr&mapExtent=605280.0000000001,149310.00000000006,701280.0000000001,186810.00000000006)

[Before](https://api3.geo.admin.ch/rest/services/ech/MapServer/ch.bav.haltestellen-oev/8573140/htmlPopup?coord=688630.0000000001,165610.00000000006&imageDisplay=1920,750,96&lang=fr&mapExtent=605280.0000000001,149310.00000000006,701280.0000000001,186810.00000000006)

[Test Link](https://mf-chsdi3.dev.bgdi.ch/iframe_service/shorten/6f2235f245)

### CODE REVIEW

File: chsdi/templates/htmlpopup/oev_haltestellen.mako [hunk 1]
```diff
@@ -219,10 +213,16 @@ $(document).ready(function() {
...
 <br />
 <div>
  <iframe src="${''.join((c['baseUrl'], '/embed.html', '?', c['layerBodId'], '=', str(c['featureId']), '&lang=', lang, '&topic=', topic))}" width='580' height='300' style="width: 100%;" frameborder='0' style='border:0'></iframe>
 </div>
 </%def>
+
+<%def name="extended_resources(c, lang)">
+  <link rel="stylesheet" href="//map.geo.admin.ch/master/2afd257/1610191614/1610191614/style/app.css">
+  <script src="${h.versioned(request.static_url('chsdi:static/js/jquery.min.js'))}"></script>
+  <script src="//map.geo.admin.ch/master/2afd257/1610191614/src/lib/moment-with-customlocales.min.js"></script>
```
Comment (oterral) [inline chsdi/templates/htmlpopup/oev_haltestellen.mako]: Why doe we need to load theses scripts?

Comment (loicgasser) [inline chsdi/templates/htmlpopup/oev_haltestellen.mako]: This is not optimized. The css we use for the stations tooltip are hosted in geoadmin.
Ideally we should migrate them from geoadmin.

Now I wanted to ask you. What is customed in the momentJS lib we have in geoadmin?
If it's nothing we use, we should use a CDN instead.

**Comment by gjn** [comment]
I like it in general. That's what I wanted back when we created the winddaten tooltip.

Anyhow, some problems arise, mainly the css/styling thing, which can't be driven by parent page anymore but needs to be self-contained.

If possible, we should try to embed the needed styles in the iframe service directly.

File: chsdi/templates/htmlpopup/oev_haltestellen.mako (hunk 1 above)
Comment (oterral) [inline chsdi/templates/htmlpopup/oev_haltestellen.mako]: It's just a custom build only with our languages you can use a cdn I think for jquery too.
For the css you should copy it there if there is any.

**Comment by loicgasser** [comment]
Yes I'll remove CSS that are not needed in geoadmin anymore and add them to ``extended.css``.
I'll switch to a CDN for momentjs as well.

**Comment by loicgasser** [comment]
updated there's nothing to remove in geoadmin, it's used by other components.

File: chsdi/templates/htmlpopup/oev_haltestellen.mako [hunk 2]
```diff
@@ -219,10 +216,17 @@ $(document).ready(function() {
...
 </table>
 <br />
 <div>
  <iframe src="${''.join((c['baseUrl'], '/embed.html', '?', c['layerBodId'], '=', str(c['featureId']), '&lang=', lang, '&topic=', topic))}" width='580' height='300' style="width: 100%;" frameborder='0' style='border:0'></iframe>
 </div>
 </%def>
+
+<%def name="extended_resources(c, lang)">
+  <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+  <link href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.6.4/css/bootstrap-datepicker.min.css" rel="stylesheet">
```
Comment (oterral) [inline chsdi/templates/htmlpopup/oev_haltestellen.mako]: Where is used the datepicker ?

File: chsdi/templates/htmlpopup/oev_haltestellen.mako [hunk 3]
```diff
@@ -219,10 +216,17 @@ $(document).ready(function() {
...
+  <script src="${h.versioned(request.static_url('chsdi:static/js/jquery.min.js'))}"></script>
```
Comment (oterral) [inline chsdi/templates/htmlpopup/oev_haltestellen.mako]: Can't you use a external cdn?

**Comment by loicgasser** [comment]
Adapted to comments, thanks @oterral for the review.
Merging once Jenkins is happy

**Comment by gjn** [comment]
Did it break windgeschwindigkeiten popup? I get a 500 here:

https://mf-chsdi3.dev.bgdi.ch/iframe_service/1610261737/rest/services/ech/MapServer/ch.bfe.windenergie-geschwindigkeit_h50/758_1520/htmlPopup?coord=561225.8099999999,143924.54&imageDisplay=1920,674,96&lang=fr&mapExtent=549125.81,140954.54,568325.81,147694.54

**Comment by loicgasser** [comment]
deploying branch

**Comment by gjn** [comment]
Let me know when it's ready for testing.

**Comment by loicgasser** [comment]
Ok got it, it's a boto credential error I think.

See:

http://mf-chsdi3.dev.bgdi.ch/ltgal/rest/services/ech/MapServer/ch.bfe.windenergie-geschwindigkeit_h50/758_1520/htmlPopup?coord=561225.8099999999,143924.54&imageDisplay=1920,674,96&lang=fr&mapExtent=549125.81,140954.54,568325.81,147694.54

and:

https://mf-chsdi3.dev.bgdi.ch/ltgal/rest/services/ech/MapServer/ch.bav.haltestellen-oev/8573140/htmlPopup?coord=688630.0000000001,165610.00000000006&imageDisplay=1920,750,96&lang=fr&mapExtent=605280.0000000001,149310.00000000006,701280.0000000001,186810.00000000006

**Comment by loicgasser** [comment]
jenkins test

**Comment by gjn** [comment]
Yep. Must be it.

This is very good stuff. I like it!

**Pull request merged**